### DISCUSSION TITLE
flake8 plugins using pyproject.toml (this plugin)

### DISCUSSION THREAD

**Discussion by christianparpart** [john-hen/Flake8-pyproject#23] [General]
Nice, however, @john-hen,

I am kinda new to flake8, but I have to create a flake8 plugin and it is mandatory for me to use pyproject.toml. I wasn't really successful so far. I have a standalone checker script, that's using the `ast` module to treewalk and do what I want to check for. We decided to go for flake8 rather than pylint, because pylint already is very much slower than flake8, and the `ast` module is totally sufficient to what I want to check for.

But we do not want to have any setup.cfg / setup.py files and the likes, but only pyproject.toml and have the project (here: flake8 plugin) configured using this.

I'm asking here because maybe it's already possible and I just didn't see the solution right in front of my eyes. :-)

**Comment by john-hen** [comment]
What do you mean, you weren't successful? What problem did you actually run into? I don't see how your use case is different from everyone else's: there's Python code that needs linting, and Flake8 is configured in `pyproject.toml`.

**Reply by christianparpart** [reply]
I mean, I code like this in my test case:
```python
(flake8_path / "test.py").write_text(code)
result = flake8_path.run_flake8()
assert len(result.out_lines) == expected_errors
```
And its `result.exit_code` is always 0 (success), even though sometimes `code` contains faulty code.

I figured that `run_flake8` is actually invoking the binary using process spawn rather than using flake8 as a module. And most likely it doesn't know about my plugin, because how should it know about it. I'd need to point it to some file that contains my checker.

There is the `--require-plugins FOOS` option, so it seems, but it's not giving me a way to tell flake8 where this plugin is located in.

It is not yet installed, because this is literally the flake8 plugin git repository, that I want to self-test, of course.

I now wonder how to procede from here. :)

p.s.: The flake8 plugin I am working on is [here](https://github.com/frequenz-floss/flake8-numbers/pull/4/files)

**Reply by john-hen** [reply]
It is good practice (if not "best practice", as they say) to install the package that you're testing. During development with `pip install --editable .` so that you can keep working on the code. If everything is properly set up, Flake8 then knows about your plugin the same way it would otherwise: because you've defined a Flake8 entry point in  `pyproject.toml`.

It's how this project here is set up too. The test suite assumes that `Flake8-pyproject` is already installed, and in CI we simply install it before running the tests.
